cmd/influxd/upgrade/config.go

```diff
@@ -152,7 +152,13 @@ func (cu *configUpgrader) save(config map[string]interface{}, path string) (stri
 		return "", err
 	}
 
-	err := ioutil.WriteFile(path, buf.Bytes(), 0666)
+	// Try writing config to the requested location.
```

Comment:
i wouldn't test if `err == nil`.

i'd test for `err != nil` and handle it here.

Reply:
Agreed this is non-standard. The `err != nil` case for both the `MkdirAll` and `WriteFile` cases is to fall through instead of returning `err`. I'll check if we can remove that behavior, it's confused me more times than it's helped me.